subcommand/webhook-cert-manager/command.go

```diff
@@ -0,0 +1,277 @@
+	}
+
+	go c.certWatcher(ctx, certCh, c.clientset, c.UI)
+
+	closeNotifiers := func() {
+		for _, notifier := range notifiers {
+			notifier.Stop()
+		}
+	}
+	defer closeNotifiers()
+
+	// Interrupted, gracefully exit
+	select {
+	case <-c.sigCh:
+		closeNotifiers()
+		cancelFunc()
+		return 0
+	}
+}
+
+func (c *Command) certWatcher(ctx context.Context, ch <-chan cert.MetaBundle, clientset kubernetes.Interface, log cli.Ui) {
+	var bundle cert.MetaBundle
+	for {
+		select {
+		case bundle = <-ch:
+			log.Info("updated certificate bundle received. updating webhook certs.")
+			// Bundle is updated, set it up
+
+		case <-time.After(30 * time.Minute):
+			// This forces the mutating ctrlWebhook config to remain updated
+			// fairly quickly. This is done every 30 minutes to ensure the certificates
+			// are in sync. Because the certificate and key are being read from a secret,
+			// this does not have to be processed as aggressively as the 1 sec time in
+			// the connect inject cert watcher.
+
+		case <-ctx.Done():
+			// Quit
+			return
+		}
+
+		log.Info("getting secret from kubernetes")
+		certSecret, err := clientset.CoreV1().Secrets(bundle.SecretNamespace).Get(ctx, bundle.SecretName, metav1.GetOptions{})
+		if err != nil && k8serrors.IsNotFound(err) {
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      bundle.SecretName,
+					Namespace: bundle.SecretNamespace,
+				},
+				Data: map[string][]byte{
+					corev1.TLSCertKey:       bundle.Cert,
+					corev1.TLSPrivateKeyKey: bundle.Key,
+				},
+				Type: corev1.SecretTypeTLS,
+			}
+
+			log.Info("creating kubernetes secret")
+			if _, err := clientset.CoreV1().Secrets(bundle.SecretNamespace).Create(ctx, secret, metav1.CreateOptions{}); err != nil {
+				log.Error(fmt.Sprintf("error writing secret to API server: %s", err))
+				continue
+			}
+
+			log.Info("updating webhook configuration with new CA")
+			if err := c.updateWebhookConfig(ctx, bundle, clientset); err != nil {
+				log.Error(fmt.Sprintf("error updating webhook configuration: %s", err))
+				continue
+			}
+			continue
+		} else if err != nil {
+			log.Error(fmt.Sprintf("error getting secret for API server: %s", err))
+			continue
+		}
+
+		// Don't update secret if the certificate is unchanged.
+		if bytes.Equal(certSecret.Data[corev1.TLSCertKey], bundle.Cert) {
+			continue
+		}
+
+		certSecret.Data[corev1.TLSCertKey] = bundle.Cert
+		certSecret.Data[corev1.TLSPrivateKeyKey] = bundle.Key
+
+		log.Info("updating secret with new certificate")
+		_, err = clientset.CoreV1().Secrets(bundle.SecretNamespace).Update(ctx, certSecret, metav1.UpdateOptions{})
+		if err != nil {
+			log.Error(fmt.Sprintf("error updating secret with certificate: %s", err))
+			continue
+		}
+
+		log.Info("updating webhook configuration with new CA")
+		if err := c.updateWebhookConfig(ctx, bundle, clientset); err != nil {
+			log.Error(fmt.Sprintf("error updating webhook configuration: %s", err))
+			continue
+		}
```

Comment:
I think we're already upding it above. Do we need to do it twice?

Reply:
In case of helm upgrades where the cert-manager deployment restarts, it will lead to a situation where the secret already exists so it will fall into this flow where a new CA might be generated + secret exists, in which case it is safer to update the WebhookConfig as well. But yeah, this will not be required for every new leaf certificate. thoughts?